integration/startup_test.go

```diff
@@ -137,3 +143,141 @@ func TestSnapshotterSystemdStartup(t *testing.T) {
 		})
 	}
 }
+
+func TestStartWithDefaultConfig(t *testing.T) {
+	defaultConfigToml, err := os.ReadFile(defaultConfigFileLocation)
+	if err != nil {
+		t.Fatalf("error fetching example toml")
+	}
+
+	sh, c := newSnapshotterBaseShell(t)
+	defer c()
+
+	rebootContainerd(t, sh, getContainerdConfigToml(t, false), string(defaultConfigToml))
+	// This will error internally if it fails to boot. If it boots successfully,
+	// the config was successfully parsed and snapshotter is running
+}
+
+// TestStartWithoutConfig checks that SOCI can start with specified config paths
+func TestStartWithConfigPaths(t *testing.T) {
+	sh, done := newSnapshotterBaseShell(t)
+	defer done()
+
+	snapshotterSocket := "/run/soci-snapshotter-grpc/soci-snapshotter-grpc.sock"
+
+	tests := []struct {
+		name             string
+		defaultPath      bool
+		fileExists       bool
+		expectedErrorStr string
+	}{
+		{
+			name:        "should start if config file is present in default location",
+			defaultPath: true,
+			fileExists:  true,
+		},
+		{
+			name:        "should start if config file is not present in default location",
+			defaultPath: true,
+			fileExists:  false,
+		},
+		{
+			name:        "should start if config file is present in non-default specified location",
+			defaultPath: false,
+			fileExists:  true,
+		},
+		{
+			name:             "should not start if config file is not present in non-default specified location",
+			defaultPath:      false,
+			fileExists:       false,
+			expectedErrorStr: "failed to open config file",
+		},
+	}
+
+	for _, tc := range tests {
+		t.Run(tc.name, func(t *testing.T) {
+			// rebootContainerd would be ideal here, but that always uses a config,
+			// so we just manually start SOCI with/without a config file.
+			err := testutil.KillMatchingProcess(sh, "soci-snapshotter-grpc")
+			if err != nil {
+				sh.Fatal("failed to kill soci: %v", err)
+			}
+
+			snRunCmd := []string{"/usr/local/bin/soci-snapshotter-grpc", "--log-level", sociLogLevel,
+				"--address", snapshotterSocket}
+
+			var configPath string
+			if tc.defaultPath {
+				configPath = defaultSnapshotterConfigPath
+			} else {
+				dir, err := testutil.TempDir(sh)
+				if err != nil {
+					t.Fatalf("error creating temp dir: %v", err)
+				}
+				defer func() {
+					sh.X("rm", "-rf", dir)
+				}()
+
+				configPath = filepath.Join(dir, "config.toml")
+				snRunCmd = append(snRunCmd, "--config", configPath)
+			}
+
+			sh.X("rm", "-rf", configPath)
+			if tc.fileExists {
+				sh.X("touch", configPath)
+			}
+
+			outR, errR, err := sh.R(snRunCmd...)
+			if err != nil {
+				t.Fatalf("failed to create pipe: %v", err)
+			}
+			reporter := testutil.NewTestingReporter(t)
+			m := testutil.NewLogMonitor(reporter, outR, errR)
+			errMatchCh := make(chan bool, 1)
```

Comment:
nit: not blocking, but a lot of the other tests where we do this just assign to a captured variable like:

```
var foundExpectedError bool
m.Add("find expected error", func( rawL string) {
    if strings.Contains(rawL, expectedError) {
        foundExpectedError = true
   }
})
```

Reply:
Sure, I did also just realize the channel approach means only the first fatal message will be logged, which if there's more than one we want to evaluate all of them. (Though, I don't know why this would ever happen?)